File: app/src/main/java/jsonHandler/model/JsonWeatherHandler.java
package jsonHandler.model;

import android.support.annotation.NonNull;

import org.json.JSONException;
import org.json.JSONObject;

public class JsonWeatherHandler {

    private static JSONObject getObject(String tagName, @NonNull JSONObject jsonObject) throws JSONException {
        return jsonObject.getJSONObject(tagName);
    }

    private static String getString(String tagName, @NonNull JSONObject jsonObject) throws JSONException {
        return jsonObject.getString(tagName);
    }

    private static float getFloat(String tagName, @NonNull JSONObject jsonObject) throws JSONException {
        return (float) jsonObject.getDouble(tagName);
    }

    private static int getInt(String tagName, @NonNull JSONObject jsonObject) throws JSONException {
        return jsonObject.getInt(tagName);
    }

    public static Weather getWeather(@NonNull String data)throws JSONException{

        Weather weather = new Weather();
        JSONObject jsonObject = new JSONObject(data);

        JSONObject coordinateObject =getObject("coord", jsonObject);
        weather.getLocationWeather().setLongitude(getFloat("lon", coordinateObject));
        weather.getLocationWeather().setLatitude(getFloat("lat", coordinateObject));

        JSONObject sysObject = getObject("sys", jsonObject);
        weather.getLocationWeather().setNameCountry(getString("country", sysObject));
        weather.getLocationWeather().setNameCity(getString("name", jsonObject));
        weather.getLocationWeather().setSunrise(getInt("sunrise", sysObject));
        weather.getLocationWeather().setSunset(getInt("sunset", sysObject));

        JSONObject jsonWeather = jsonObject.getJSONArray("weather").getJSONObject(0);

        weather.getCurrentCondition().setWeatherId(getInt("id", jsonWeather));
        weather.getCurrentCondition().setDescripion(getString("description", jsonWeather));
        weather.getCurrentCondition().setCondition(getString("main", jsonWeather));
        weather.getCurrentCondition().setIconWeather(getString("icon", jsonWeather));

        JSONObject jsonMainObject = getObject("main", jsonObject);

        weather.getCurrentCondition().setHumidity(getInt("humidity", jsonMainObject));
        weather.getCurrentCondition().setPressure(getInt("pressure", jsonMainObject));
        weather.getTemperature().setMaxTemperature(getFloat("temp_max", jsonMainObject));
        weather.getTemperature().setMinTemperature(getFloat("temp_min", jsonMainObject));
        weather.getTemperature().setCurrentTemperature(getFloat("temp", jsonMainObject));

        JSONObject jsonWindObject = getObject("wind", jsonObject);

        weather.getWind().setSpeed(getFloat("speed", jsonWindObject));
        weather.getWind().setDeg(getFloat("deg", jsonWindObject));

        JSONObject jsonCloudsObject = getObject("clouds", jsonObject);
        weather.getClouds().setPercent(getInt("all", jsonCloudsObject));

        return weather;
    }
}

File: app/src/main/java/jsonHandler/jsonWeatherHandler.kt
package jsonHandler

import org.json.JSONException
import org.json.JSONObject

class jsonWeatherHandler {
    companion object{
        @Throws(JSONException::class)
        fun getWeather(data:String): Weather? {

            var weather = Weather()
            var jsonObject = JSONObject(data)
            val location = LocationWeather()
            val coordObject:JSONObject = getObject("coord", jsonObject)
            location.latitude = getFloat("lat", coordObject)
            location.longitude = getFloat("lon", coordObject)

            val sysObjects = getObject("sys", jsonObject)
            location.nameCounty = getString("country", sysObjects)
            location.sunrise = getInt("sunrise", sysObjects).toLong()
            location.sunset = getInt("sunset", sysObjects).toLong()
            location.nameCity = getString("name", sysObjects)

            weather.location = location

            val jsonArray = jsonObject.getJSONArray("weather")

            val jsonWeather = jsonArray.getJSONObject(0)

            weather.currentCondidtion.weatherId = getInt("id", jsonWeather)
            weather.currentCondidtion.descripion = getString("description", jsonWeather)
            weather.currentCondidtion.condition = getString("main", jsonWeather)
            weather.currentCondidtion.iconWeather = getString("icon", jsonWeather)

            val mainJson = getObject("main", jsonObject)

            weather.currentCondidtion.humidity = getInt("humidity", mainJson).toFloat()
            weather.currentCondidtion.pressure = getInt("pressure", mainJson).toFloat()
            weather.temperature.maxTemperature = getFloat("temp_max", mainJson)
            weather.temperature.minTemperature = getFloat("temp_min", mainJson)
            weather.temperature.temperature = getFloat("temp", mainJson)

            val windObj = getObject("wind", jsonObject)
            weather.wind.speed = getFloat("speed", windObj)
            weather.wind.deg = getFloat("deg", windObj)

            val cloudObj = getObject("clouds", jsonObject)
            weather.clouds.percent = getInt("all", cloudObj)



            return weather
        }

        private fun getObject(tagName:String, jsonObject: JSONObject): JSONObject {
            return jsonObject.getJSONObject(tagName)
        }

        private fun getString(tagName:String, jsonObject: JSONObject): String{
            return jsonObject.getString(tagName)
        }

        private fun getFloat(tagName: String, jsonObject: JSONObject):Float{
            return jsonObject.getDouble(tagName).toFloat()
        }

        private fun getInt(tagName: String, jsonObject: JSONObject): Int{
            return jsonObject.getInt(tagName)
        }
    }
}